File: JavaScript/Week-02/Day-1/DiagonalMatrix.ts
// -  Create (dynamically*) a two dimensional list
//    with the following matrix**. Use a loop!
//
//    0 0 0 1
//    0 0 1 0
//    0 1 0 0
//    1 0 0 0
//
// -  Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array

let matrixsize: number = 4;
let matrix: number[][] = [];

for (let x: number = 0; x < matrixsize; x++) {
  matrix.push([]);
  for (let y: number = 0; y < matrixsize; y++) {
    if (y === (matrixsize - x - 1)) {
      matrix[x].push(1);
    } else {
      matrix[x].push(0);
    }
  }
  console.log(matrix[x]);
}

File: Java/Week1/Day-4-5/DiagonalMatrix.java
// - Create (dynamically) a two dimensional array
// with the following matrix. Use a loop!
//
// 1 0 0 0
// 0 1 0 0
// 0 0 1 0
// 0 0 0 1
//
// - Print this two dimensional array to the output

/**
 * DiagonalMatrix
 */
public class DiagonalMatrix {

  public static void main(String[] args) {
    int size = 4;
    int[][] matrix = new int[size][size];

    for (int outer = 0; outer < size; outer++) {
      for (int inner = 0; inner < size; inner++) {
        if (outer == inner) {
          matrix[outer][inner] = 1;
        } else {
          matrix[outer][inner] = 0;
        }
      }
    }
    for (int outer = 0; outer < size; outer++) {
      for (int inner = 0; inner < size; inner++) {
        System.out.print(matrix[outer][inner] + " ");
      }
      System.out.println();;
    }
  }
}